### PR TITLE
[Keyboard] Add handwired BDN9-BLE

### PR DESCRIPTION
**Pull request by Merlin04**
## Description

Add BDN9-BLE, a handwired BDN9 macropad modified to add Bluetooth. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/bdn9_ble/readme.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# BDN9-BLE
+
+![BDN9](https://cdn.shopify.com/s/files/1/1851/5125/products/image_bd8d9423-950e-4aad-bea5-665d896f879a_530x@2x.jpg?v=1547909493)
+
+A handwired modified version of the Keebio BDN9 macropad to support BLE through the Adafruit Feather 32u4 Bluefruit LE.
+
+Keyboard Maintainer: [merlin04/KeyPCB](https://github.com/merlin04)
```
Comment (fauxpark) [inline keyboards/handwired/bdn9_ble/readme.md]:
```suggestion
* Keyboard Maintainer: [merlin04/KeyPCB](https://github.com/merlin04)
```

File: keyboards/handwired/bdn9_ble/readme.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# BDN9-BLE
+
+![BDN9](https://cdn.shopify.com/s/files/1/1851/5125/products/image_bd8d9423-950e-4aad-bea5-665d896f879a_530x@2x.jpg?v=1547909493)
+
+A handwired modified version of the Keebio BDN9 macropad to support BLE through the Adafruit Feather 32u4 Bluefruit LE.
+
+Keyboard Maintainer: [merlin04/KeyPCB](https://github.com/merlin04)  
+Hardware Supported: Adafruit Feather 32u4 Bluefruit LE
```
Comment (fauxpark) [inline keyboards/handwired/bdn9_ble/readme.md]:
```suggestion
* Hardware Supported: Adafruit Feather 32u4 Bluefruit LE
```

File: keyboards/handwired/bdn9_ble/rules.mk [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
```
Comment (fauxpark) [inline keyboards/handwired/bdn9_ble/rules.mk]:
```suggestion
# Processor frequency
F_CPU = 8000000

# Bootloader selection
```

File: keyboards/handwired/bdn9_ble/rules.mk [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = caterina
+F_CPU = 8000000
```
Comment (fauxpark) [inline keyboards/handwired/bdn9_ble/rules.mk]:
```suggestion
```

File: keyboards/handwired/bdn9_ble/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+#define ______ KC_TRNS
```
Comment (fauxpark) [inline keyboards/handwired/bdn9_ble/keymaps/default/keymap.c]:
```suggestion
```

File: keyboards/handwired/bdn9_ble/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    #endif
+    /*
+     * ------------------------------------------------------------------------
+     * |                | Automatic BLE output detection | Reset               |
+     * | Backlight Step | Use USB port for output        |                     |
+     * | Backlight Tgl  | Use Bluetooth for output       | (currently pressed) |
+     * ------------------------------------------------------------------------
+     */
+    [1] = LAYOUT(
+        ______,  OUT_AUTO, RESET,
```
Comment (fauxpark) [inline keyboards/handwired/bdn9_ble/keymaps/default/keymap.c]:
```suggestion
        _______, OUT_AUTO, RESET,
```

File: keyboards/handwired/bdn9_ble/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        BL_STEP, OUT_USB,  ______,
```
Comment (fauxpark) [inline keyboards/handwired/bdn9_ble/keymaps/default/keymap.c]:
```suggestion
        BL_STEP, OUT_USB, _______,
```

File: keyboards/handwired/bdn9_ble/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        BL_TOGG, OUT_BT,   ______
```
Comment (fauxpark) [inline keyboards/handwired/bdn9_ble/keymaps/default/keymap.c]:
```suggestion
        BL_TOGG, OUT_BT,  _______
```

File: keyboards/handwired/bdn9_ble/rules.mk (hunk 4 above)
Comment (Merlin04) [reply keyboards/handwired/bdn9_ble/rules.mk]: Why should this line be removed?

Comment (fauxpark) [reply keyboards/handwired/bdn9_ble/rules.mk]: The suggestion above adds it before the bootloader section.

File: keyboards/handwired/bdn9_ble/rules.mk [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+BOOTMAGIC_ENABLE = no      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality on B7 by default
```
Comment (fauxpark) [inline keyboards/handwired/bdn9_ble/rules.mk]:
```suggestion
BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality
```

**Comment by drashna** [comment]
Just to make sure, this is a BDN9 with a bluetooth module (and battery)?   Otherwise, it's identical hardware?

**Comment by noroadsleft** [comment]
Didn't see @drashna's question until after approval; will leave judgement on when to merge this to him.

**Comment by zvecr** [comment]
Potentially this could be just a keymap like <https://github.com/qmk/qmk_firmware/tree/master/keyboards/40percentclub/mf68/keymaps/mf68_ble>

**Comment by Merlin04** [comment]
It's a BDN9 PCB, but an Adafruit Feather 32u4 Bluefruit LE has been connected. It does not use the same pins.

**Comment by drashna** [comment]
I did notice that (after commenting).  Though, if this is using the original PCB ... you could just undef the pins in your config.h, and redefine them.   That would remove the need to specify a new boards

And you can override the MCU speed in your keymap's rules.mk file. 

Personally, I feel that this would be a better way to handle this, but I'm not opposed to merging this, as is.

**Comment by Merlin04** [comment]
I think that it would work as a keymap, but I'd prefer a new keyboard because that would let me use the Configurator with it.

**Pull request merged**